src/change-location-link/change-location-link.component.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import Button from "carbon-components-react/es/components/Button";
+import { useTranslation } from "react-i18next";
+import styles from "./change-location.link.component.scss";
+import Location20 from "@carbon/icons-react/es/location/20";
+import { navigate } from "@openmrs/esm-framework";
+const openmrsSpaBase = window["getOpenmrsSpaBase"]();
+
+interface ChangeLocationLinkProps {
+  referer?: string;
+  currentLocation: string;
+}
+
+const ChangeLocationLink: React.FC<ChangeLocationLinkProps> = ({
+  referer,
+  currentLocation,
+}) => {
+  const { t } = useTranslation();
+
+  const changeLocation = () => {
+    navigate({ to: `${openmrsSpaBase}login/location?returnToUrl=${referer}` });
```

Comment:
I think it would be better to use the template variable instead of a template string. This way you don't need to worry / care about getting openmrsSpaBase from `window`.

Here are the variables:
https://github.com/openmrs/openmrs-esm-core/blob/master/packages/esm-config/src/navigation/interpolate-string.ts#L14